client/main/dataset.js

```diff
@@ -444,10 +448,20 @@ const DataSetModel = Backbone.Model.extend({
             levels = new Array(columnPB.levels.length);
             for (let i = 0; i < levels.length; i++) {
                 let levelPB = columnPB.levels[i];
-                levels[i] = {
-                    label: levelPB.label,
-                    value: levelPB.value,
-                };
+                if (column.measureType === 'nominaltext') {
+                    levels[i] = {
+                        label: levelPB.label,
+                        value: i,
+                        importValue: levelPB.importValue
+                    };
+                }
+                else {
+                    levels[i] = {
+                        label: levelPB.label,
+                        value: levelPB.value,
+                        importValue: levelPB.value.toString()
```

Comment:
My theory here is that doing it this way means that the check of measure type only has to happen on the client side once and from here everything works the same. I feel like the passing around on the client side is not heavy in any stretch but allows for consistence. When it comes to the protobuffers it's obviously a different story.